packages/form-plugin/src/form.plugin.ts

```diff
@@ -24,13 +26,18 @@ export class NgxsFormPlugin implements NgxsPlugin {
 
     let nextState = state;
 
-    if (type === UpdateFormValue.type || type === UpdateForm.type) {
+    if (type === UpdateFormValue.type || type === UpdateForm.type || type === ResetForm.type) {
       const { value } = event.payload;
       const payloadValue = Array.isArray(value) ? value.slice() : { ...value };
       const path = this.joinPathWithPropertyPath(event);
       nextState = setValue(nextState, path, payloadValue);
     }
 
+    if (type === ResetForm.type) {
+      const model = getValue(nextState, `${event.payload.path}.model`);
+      nextState = setValue(nextState, `${event.payload.path}`, { model: model });
```

Comment:
What is the intention of this code here?
To wipe out the other properties for the form?
It would wipe out the `status`, `errors`, `dirty` and `disabled` properties. Not sure if this is what we want.
Surely all of the properties will be reset when the `form.reset` happens, or maybe the `form.resetForm` would make this not needed anymore. Maybe after making that change remove this and see if you get the behaviour you are intending.

Reply:
You alright. I have to work on it a little bit. We can update form state after from.reset called in directive like debounce change subscriptions.

I think we create new function for update state and use it in debounce subscription and after form reset.

```typescript
  updateFormStateWithRawValue() {
    const value = this._formGroupDirective.control.getRawValue();
    this._updating = true;
    this._store
      .dispatch([
        new UpdateFormValue({
          path: this.path,
          value
        }),
        new UpdateFormDirty({
          path: this.path,
          dirty: this._formGroupDirective.dirty
        }),
        new UpdateFormErrors({
          path: this.path,
          errors: this._formGroupDirective.errors
        })
      ])
      .subscribe({
        error: () => (this._updating = false),
        complete: () => (this._updating = false)
      });
  }
```
Debounce Change
```typescript
this._formGroupDirective.valueChanges!.pipe(this.debounceChange()).subscribe(() => {
      this.updateFormStateWithRawValue();
});
```
After Form Reset
```typescript
    this._actions$
      .pipe(
        ofActionDispatched(ResetForm),
        filter((action: ResetForm) => action.payload.path === this.path),
        takeUntil(this._destroy$)
      )
      .subscribe(({ payload: { value } }: ResetForm) => {
        this.form.reset(value);
        this.updateFormStateWithRawValue();
        this._cd.markForCheck();
      });
```

How about that?

Reply:
This code worked correctly. I think it was better like this. I will commit my changes if it is suitable for you.

Reply:
Is this discussion resolved? I think that this was covered by later changes and tests?

Reply:
Yes this discussion resolved. This is required and this is covered with tests. Because if we are reset form state without form and ngxsForm directive this code block resets our state.

I asked this case, if i understood correctly this is your answer:

> My gut feel is that if you are not bound with an `ngxsForm` then the model and everything else should be reset. ie. not dirty, not touched, no errors, reset status, etc.